engine/dlib/wscript

```diff
@@ -40,6 +42,10 @@ def configure(conf):
     conf.env.append_value('LIBPATH', build_util.get_dynamo_ext('lib', build_util.get_target_platform()))
     conf.env.append_unique('DEFINES', 'DLIB_LOG_DOMAIN="DLIB"')
 
+    if target_os != TargetOS.WINDOWS:
+        conf.env.append_value('CFLAGS', ["-fdebug-compilation-dir=engine/resource"])
+        conf.env.append_value('CXXFLAGS', ["-fdebug-compilation-dir=engine/resource"])
```

Comment:
An alternative would be to add this to waf_dynamo, as part of one of the "@feature` hooks instead
E.g. checking the output folder of a TaskGen and creating the path dynamically.

In its current state, it feels easy to forget or miss the nuance if it gets left out (e.g. when creating a new library)

Reply:
Changes https://github.com/defold/defold/pull/10369#discussion_r2024630854